src/kernel/capinv.c

```diff
@@ -569,9 +577,8 @@ capinv_int_snd(struct thread *rcv_thd, struct pt_regs *regs)
 	assert(rcv_tcap);
 
 	next = asnd_process(rcv_thd, thd, rcv_tcap, tcap);
-	if (next == thd) return 0;
+	if (next == thd) return 1; /* current thread is a preempted one! */
```

Comment:
Yeah I have that doubt. When I kept it as 'return 0', it was page-faulting. Anyway, I'll check further into this, this is also useful in the other comment where you suggested to just do jmp restore_from_thd. I'll check that! Thanks!